lib/blockscore/actions/retrieve.rb

```diff
@@ -16,7 +16,7 @@ module ClassMethods
         def retrieve(id, options = {})
           fail ArgumentError, 'ID must be supplied' if id.nil? || id.empty?
           fail ArgumentError, 'ID is malformed' unless id =~ RESOURCE_ID_FORMAT
-          req = ->() { get("#{endpoint}/#{id}", options) }
+          req = ->() { get(endpoint + id, options) }
           new(id: id, &req)
```

Comment:
> That is a construction I was not aware ot

Yeah, it's not *exactly* as straight-forward as this, but it's pretty close:

```diff
diff --git a/lib/blockscore/actions/retrieve.rb b/lib/blockscore/actions/retrieve.rb
index 5aac386..9d1c7e6 100644
--- a/lib/blockscore/actions/retrieve.rb
+++ b/lib/blockscore/actions/retrieve.rb
@@ -16,8 +16,7 @@ module BlockScore
         def retrieve(id, options = {})
           fail ArgumentError, 'ID must be supplied' if id.nil? || id.empty?
           fail ArgumentError, 'ID is malformed' unless id =~ RESOURCE_ID_FORMAT
-          req = ->() { get(endpoint + id, options) }
-          new(id: id, &req)
+          new(id: id) { |object| get(object.member_endpoint, options) }
         end
       end
 
diff --git a/lib/blockscore/base.rb b/lib/blockscore/base.rb
index e2cddad..13c14e2 100644
--- a/lib/blockscore/base.rb
+++ b/lib/blockscore/base.rb
@@ -95,8 +95,9 @@ module BlockScore
     end
 
     def force!
-      @attributes = @proc.call.attributes.merge(@attributes)
       @loaded = true
+      args = @proc.arity.zero? ? [] : [self]
+      @attributes = @proc.call(*args).attributes.merge(@attributes)
     end
 
     def method_missing(method, *args, &block)
```

The reason for moving `@loaded` earlier is to break an infinite loop that can occur when the block references `attributes` (which are not yet loaded, which triggers `#force!`, etc)